### ISSUE TITLE
vkSetDebugUtilsObjectNameEXT does not copy contents of VkDebugUtilsObjectNameInfoEXT::pObjectName string

### ISSUE THREAD

**Issue by Didgy74** [KhronosGroup/Vulkan-ValidationLayers#1168]
This seems to have broken in a recent update. Testing in a Vulkan Community Discord Server showed that this functionality behaved correctly in SDK version 1.1.108.

In 1.1.144, when setting an object name using `vkSetDebugUtilsObjectNameEXT` the validation layer does not seem to copy the contents of the string supplied, but rather purely referencing the inputted string. I have written some example code to trigger and show the problem: https://pastebin.com/CQgSB0SS

The relevant code starts at line #72 until the end of the file:
```cpp
    std::string handleName = "ORIGINAL HANDLE NAME";
 
    VkDebugUtilsObjectNameInfoEXT objectNameInfo{};
    objectNameInfo.sType = VK_STRUCTURE_TYPE_DEBUG_UTILS_OBJECT_NAME_INFO_EXT;
    objectNameInfo.objectType = VK_OBJECT_TYPE_DEBUG_UTILS_MESSENGER_EXT;
    objectNameInfo.objectHandle = (uint64_t)deviceHandle;
    objectNameInfo.pObjectName = handleName.data();
    PFN_vkSetDebugUtilsObjectNameEXT setDebugUtilsObjectName = (PFN_vkSetDebugUtilsObjectNameEXT)vkGetInstanceProcAddr(instanceHandle, "vkSetDebugUtilsObjectNameEXT");
    if (setDebugUtilsObjectName == nullptr)
        return 6;
    VkResult setNameResult = setDebugUtilsObjectName(deviceHandle, &objectNameInfo);
    if (setNameResult != VK_SUCCESS)
        return 7;
 
    // Modifying our original memory makes the validation layer's object name be altered also.
    handleName = "CHANGED HANDLE NAME";
 
    vkDestroyInstance(instanceHandle, nullptr);
 
    return 0;
}
```

The VkDevice created here should have the name "ORIGINAL HANDLE NAME". However, when running the code on Windows and Manjaro, the code produces the following validation layer output:
```
OBJ ERROR : VK_DEBUG_REPORT_OBJECT_TYPE_DEVICE_EXT object VkDevice 0x55db7b07d150[CHANGED HANDLE NAME] has not been destroyed.
OBJ ERROR : For VkDevice 0x55db7b07d150[CHANGED HANDLE NAME], VkDebugUtilsMessengerEXT 0x1[] has not been destroyed. The Vulkan spec states: All child objects created using instance must have been destroyed prior to destroying instance (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkDestroyInstance-instance-00629)
Debug messengers not removed before DestroyInstance
```

Sometimes the validation layer will show the name "CHANGED HANDLE NAME", nothing at all, or it will output garbage memory until it hits a null-byte.

**Comment by jeffbolznv** [comment]
Fixed by #1149?

**Comment by Didgy74** [comment]
Yes, that looks like it should fix it.
